retrievalmarket/impl/blockio/traverser.go

```diff
@@ -93,22 +93,28 @@ func (t *Traverser) Start(ctx context.Context) {
 				return response.input, response.err
 			}
 		}
-		nd, err := t.root.Load(ctx, ipld.LinkContext{}, chooser(t.root, ipld.LinkContext{}), loader)
+		style, err := chooser(t.root, ipld.LinkContext{})
 		if err != nil {
 			t.writeDone(ctx)
 			return
 		}
-
+		builder := style.NewBuilder()
+		err = t.root.Load(ctx, ipld.LinkContext{}, builder, loader)
+		if err != nil {
+			t.writeDone(ctx)
+			return
+		}
+		nd := builder.Build()
 		sel, err := selector.ParseSelector(t.selector)
 		if err != nil {
 			t.writeDone(ctx)
 			return
 		}
 		_ = traversal.Progress{
 			Cfg: &traversal.Config{
-				Ctx:                    ctx,
-				LinkLoader:             loader,
-				LinkNodeBuilderChooser: chooser,
+				Ctx:                        ctx,
+				LinkLoader:                 loader,
+				LinkTargetNodeStyleChooser: chooser,
```

Comment:
Nit: can this be a shortened at all?

Reply:
File under: requests for @warpfork

Reply:
I confess I hate this one too, but it does what it says on the tin.  I haven't thought of a shorter name.